src/Core/src/Platform/Windows/MauiWinUIWindow.cs

```diff
@@ -188,9 +189,15 @@ internal void UpdateTitleOnCustomTitleBar()
 				fe.GetDescendantByName<TextBlock>("AppTitle") is TextBlock tb)
 			{
 				tb.Text = Title;
+				UpdateSemanticsDescription(tb, Title);
 			}
 		}
 
+		void UpdateSemanticsDescription(FrameworkElement frameworkElement, string description)
+		{
+			AutomationProperties.SetName(frameworkElement, description);
```

Comment:
The native Window is not a DependencyObject, so we must apply the AutomationProperties to the child element of the Window header (title). On the other hand, a window implements the `IWindow` interface that inherits directly from `IElement` (not IView) so it does not have a Semantic property. In order to use Semantics in the Window we would need to make changes to the public API, for example moving the Semantic property to IElement (which would also apply these properties to other elements like Toolbar etc).